File: exercise_02/my3DDatatypes/IPoint.java
package my3DDatatypes;
public class IPoint {
	public final static float epsilon = 0.0001f;
	public Triangle triangle;
	public Vec3D ipoint;
	public float dist;

	public IPoint(Triangle tt, Vec3D ip, float d) {
		triangle = tt;
		ipoint = ip;
		dist = d;
	}
}

File: exercise_02_cpp/IPoint.cpp
#include "IPoint.h"

namespace my3DDatatypes {

	IPoint::IPoint() {
		dist = -1;
	}

	IPoint::IPoint(const Triangle &tt, const Vec3D &ip, float d) {
		triangle = tt;
		ipoint = ip;
		dist = d;
	}

}
